src/applications/disability-benefits/all-claims/tests/utils/on-submit/purge-toxic-exposure-data.unit.spec.jsx

```diff
@@ -0,0 +1,1122 @@
+              guam: false,
+              johnston: false,
+              koreandemilitarizedzone: false,
+              laos: false,
+              c123: false,
+              thailand: false,
+              vietnam: false,
+              none: false,
+            },
+            otherHerbicideLocations: {
+              description: 'This should be removed - all herbicides are false',
+              startDate: '1973-06-01',
+              endDate: '1974-12-31',
+            },
+          },
+        };
+
+        const result = purgeToxicExposureData(formData);
+
+        expect(result.toxicExposure).to.not.have.property(
+          'otherHerbicideLocations',
+        );
+        expect(result.toxicExposure).to.not.have.property('herbicide');
+      });
+    });
+
+    describe('edge cases', () => {
+      it('should return unchanged for invalid toxicExposure types (string, array, etc)', () => {
+        const stringCase = {
+          disability526ToxicExposureOptOutDataPurge: true,
+          toxicExposure: 'not-an-object',
+        };
+        const arrayCase = {
+          disability526ToxicExposureOptOutDataPurge: true,
+          toxicExposure: [],
+        };
+
+        expect(purgeToxicExposureData(stringCase)).to.deep.equal(stringCase);
+        expect(purgeToxicExposureData(arrayCase)).to.deep.equal(arrayCase);
+      });
+
+      it('should handle null and undefined values', () => {
+        const formData = {
+          disability526ToxicExposureOptOutDataPurge: true,
+          toxicExposure: {
+            conditions: { asthma: true },
+            gulfWar1990: null,
+            herbicide: undefined,
+            otherExposures: { asbestos: true },
+          },
+        };
+
+        const result = purgeToxicExposureData(formData);
+
+        expect(result.toxicExposure).to.have.property('conditions');
+        expect(result.toxicExposure).to.have.property('otherExposures');
+        expect(result.toxicExposure).to.not.have.property('gulfWar1990');
+        expect(result.toxicExposure).to.have.property('herbicide');
+        expect(result.toxicExposure.herbicide).to.be.undefined;
+      });
+
+      it('should handle empty conditions object', () => {
+        const formData = {
+          disability526ToxicExposureOptOutDataPurge: true,
+          toxicExposure: {
+            conditions: {},
+            gulfWar1990: { bahrain: true },
+          },
+        };
+
+        const result = purgeToxicExposureData(formData);
+
+        expect(result.toxicExposure.conditions).to.deep.equal({});
+        expect(result.toxicExposure.gulfWar1990).to.exist;
+      });
+
+      it('should handle missing conditions', () => {
+        const formData = {
+          disability526ToxicExposureOptOutDataPurge: true,
+          toxicExposure: {
+            gulfWar1990: { bahrain: true },
+          },
+        };
+
+        const result = purgeToxicExposureData(formData);
+
+        expect(result.toxicExposure.gulfWar1990).to.exist;
```

Comment:
What would the value be of `result.toxicExposure.gulfWar1990` if it was not associated with a condition?